doc/source/reference/array_api.rst

```diff
@@ -0,0 +1,795 @@
+       if NumPy's scalars deviate from the spec 0-D array behavior in any ways
+       other than the ones outlined in
+       :ref:`array_api-type-promotion-differences`.
+   * - ``bool()``, ``int()``, and ``float()`` only work on 0-D arrays.
+     - **Strictness**
+     - See https://github.com/numpy/numpy/issues/10404.
+   * - ``__imatmul__``
+     - **Compatible**
+     - ``np.ndarray`` does not currently implement ``__imatmul``. Note that
+       ``a @= b`` should only defined when it does not change the shape of
+       ``a``.
+   * - The ``mT`` attribute for matrix transpose.
+     - **Compatible**
+     - See `the spec definition
+       <https://data-apis.org/array-api/latest/API_specification/generated/signatures.array_object.array.mT.html>`__
+       for ``mT``.
+   * - The ``T`` attribute should error if the input is not 2-dimensional.
+     - **Breaking**
+     - See `the note in the spec
+       <https://data-apis.org/array-api/latest/API_specification/generated/signatures.array_object.array.T.html>`__.
+   * - New method ``to_device`` and attribute ``device``
+     - **Compatible**
+     - The methods would effectively not do anything since NumPy is CPU only
+
+Creation Functions Differences
+------------------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Feature
+     - Type
+     - Notes
+   * - ``copy`` keyword argument to ``asarray``
+     - **Compatible**
+     -
+   * - New ``device`` keyword argument to all array creation functions
+       (``asarray``, ``arange``, ``empty``, ``empty_like``, ``eye``, ``full``,
+       ``full_like``, ``linspace``, ``ones``, ``ones_like``, ``zeros``, and
+       ``zeros_like``).
+     - **Compatible**
+     - ``device`` would effectively do nothing, since NumPy is CPU only.
+
+Elementwise Functions Differences
+---------------------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Feature
+     - Type
+     - Notes
+   * - Various functions have been renamed.
+     - **Compatible**
+     - See :ref:`array_api-name-changes`.
+   * - Elementwise functions are only defined for given input type
+       combinations.
+     - **Strictness**
+     - See :ref:`array_api-type-strictness`.
+   * - ``bitwise_left_shift`` and ``bitwise_right_shift`` are only defined for
+       ``x2`` nonnegative.
+     - **Strictness**
+     -
+   * - ``ceil``, ``floor``, and ``trunc`` return an integer with integer
+       input.
+     - **Breaking**
+     - ``np.ceil``, ``np.floor``, and ``np.trunc`` return a floating-point
+       dtype on integer dtype input.
+
+.. _array_api-linear-algebra-differences:
+
+Linear Algebra Differences
+--------------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Feature
+     - Type
+     - Notes
+   * - ``cholesky`` includes an ``upper`` keyword argument.
+     - **Compatible**
+     -
+   * - ``cross`` does not broadcast its arguments.
+     - ???
+     -
+   * - ``cross`` does not allow size 2 vectors (only size 3).
+     - ???
+     -
```

Comment:
I'm still unclear whether these `cross` differences are a hard requirement in the spec or not. CC @rgommers @kgryte

Reply:
I've opened an issue concerning `linalg.cross` behavior: https://github.com/data-apis/array-api/issues/415. The outcome of that discussion should have bearing on potential backward compatibility concerns.

Reply:
As discussed in https://github.com/data-apis/array-api/issues/415, allowing size 2 vectors is likely to never be supported in the specification given difficulties such support presents in, e.g., JIT compilation (e.g., https://github.com/numpy/numpy/issues/13718). Accordingly, we should mark this NumPy behavior as backward incompatible with the spec.

Re: broadcasting. My sense is that we may be moving toward alignment on NumPy broadcasting behavior, but this is still TBD. Atm, I don't think we need to mark NumPy's current broadcasting behavior as backward incompatible.